modules/emacs/hm.nix

```diff
@@ -62,6 +65,9 @@ with config.stylix.fonts;
         ;; Set font
         (set-face-attribute 'default t :font "${monospace.name}" )
         ;; -----------------------------
+        ;; set opacity
+        (add-to-list 'default-frame-alist '(alpha . (${emacsOpacity} . ${emacsOpacity})))
```

Comment:
Note that there's `alpha-background` available [from emacs 29 onwards](https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=5c87d826201a5cae242ce5887a0aa7e24ad6f5ee). Might be the preferable way of setting this, but emacs 28 is still in use by many users.

The main difference is that `alpha` makes the whole window transparent (even the font of your text), and `alpha-background` only the background.

Reply:
it may be best to check the package version and use the preferred method, so those using 28 still have at least some kind of transparency. im not a heavy emacs user as most of my understanding comes from using it for various theorem provers and school, so thanks for pointing that out. i will get to implementing this later today.